reproducibility/codalab/util/analysis_utils.py

```diff
@@ -0,0 +1,124 @@
+
+        for result_type in RESULT_TYPES:
+            if len(result_dfs) > 0 and result_type in result_dfs[0]:
+                compiled_results[algorithm][result_type] = dict()
+                for metric in metrics:
+                    compiled_results[algorithm][result_type][metric] = []
+
+        for result_df in result_dfs:
+            for result_type in RESULT_TYPES:
+                if result_type not in result_df:
+                    continue
+
+                for metric in metrics:
+                    val_split = "id_val_eval" if in_distribution_val else "val_eval"
+                    result = get_early_stopped_row(
+                        result_df[result_type],
+                        result_df[val_split],
+                        sort_metrics=metrics,
+                    )
+                    compiled_results[algorithm][result_type][metric].append(
+                        result[metric]
+                    )
+
+        for result_type in RESULT_TYPES:
+            if result_type in compiled_results[algorithm]:
+                for metric in metrics:
+                    # Take the sample standard deviation (delta degree of freedom = 1)
+                    # and mean across replicates for each metric.
+                    compiled_results[algorithm][result_type][f"{metric}_std"] = np.std(
+                        compiled_results[algorithm][result_type][metric], ddof=1
+                    )
+                    compiled_results[algorithm][result_type][metric] = np.mean(
+                        compiled_results[algorithm][result_type][metric]
+                    )
+    return compiled_results
+
+
+def get_early_stopped_row(src_df, val_df, sort_metrics):
+    """
+    Get the early stopped row with the highest metric value.
+    """
+    # Sort in descending order so larger metric value the better
+    epoch = val_df.sort_values(sort_metrics, ascending=False).iloc[0]["epoch"]
+    selected_row = src_df.loc[src_df["epoch"] == epoch]
+    assert selected_row.shape[0] == 1
+    return selected_row.iloc[0]
+
+
+def get_metrics(dataset):
+    """
+    Returns a list of metrics for the given dataset. The first metric of the list
+    is used to determine the best hyperparameters on a validation set.
+
+    Parameters:
+        dataset (str): Dataset name or a string containing the dataset name.
+    """
+    if "amazon" in dataset:
+        return ["10th_percentile_acc", "acc_avg"]
+    elif "camelyon17" in dataset:
+        return ["acc_avg"]
+    elif "civilcomments" in dataset:
+        return ["acc_wg", "acc_avg"]
+    elif "fmow" in dataset:
+        return ["acc_worst_region", "acc_avg"]
+    elif "iwildcam" in dataset:
+        return ["F1-macro_all", "acc_avg"]
+    elif "ogb-molpcba" in dataset:
+        return ["ap"]
+    elif "poverty" in dataset:
+        return ["r_wg", "r_all", "r_urban:0", "r_urban:1"]
+    elif "py150" in dataset:
+        return ["acc"]
```

Comment:
To reflect the eval update on py150 code, can you update this to 'Acc (Class-Method)'?